galaxy_ng/app/api/v3/viewsets/collection.py

```diff
@@ -87,21 +92,66 @@ class CollectionViewSet(api_base.LocalSettingsMixin,
     serializer_class = CollectionSerializer
 
 
-class UnpaginatedCollectionVersionViewSet(api_base.LocalSettingsMixin,
-                                          ViewNamespaceSerializerContextMixin,
-                                          pulp_ansible_views.UnpaginatedCollectionVersionViewSet):
+class UnpaginatedCollectionVersionViewSet(
+    api_base.LocalSettingsMixin,
+    ViewNamespaceSerializerContextMixin,
+    DestroyModelMixin,
+    pulp_ansible_views.UnpaginatedCollectionVersionViewSet,
+):
     pagination_class = None
     serializer_class = UnpaginatedCollectionVersionSerializer
     permission_classes = [access_policy.CollectionAccessPolicy]
 
 
+def get_dependents(parent):
+    """Given a parent collection, return a list of collections that depend on it."""
+    key = f"{parent.namespace}.{parent.name}"
+    dependents = []
+    for child in CollectionVersion.objects.filter(dependencies__has_key=key):
+        spec = SpecifierSet(child.dependencies[key])
+        if Version(parent.version) in spec:
+            dependents.append(child)
```

Comment:
@rochacbruno in working this PR, I plan to change the version specificer comparison library from `packaging` to `semantic_version` which we use in `galaxy-importer` and a bit in `galaxy_ng`.

For example we have existing dependencies in the format of `ansible.netcommon: '1.0.0'` but for that case it looks like packaging expects `ansible.netcommon: '==1.0.0'`